File: square/src/main/java/com/raxdenstudios/square/lifecycle/ApplicationLifecycle.java
package com.raxdenstudios.square.lifecycle;

import android.content.res.Configuration;

/**
 * Created by Ángel Gómez
 *
 * Contract that defines the Application life cycle used by interceptors.
 */
public interface ApplicationLifecycle {

    void onConfigurationChanged(Configuration newConfig);

    void onCreate();

    void onLowMemory();

    void onTrimMemory(int level);

}

File: square/src/main/java/com/raxdenstudios/square/lifecycle/ApplicationLifecycle.kt
package com.raxdenstudios.square.lifecycle

import android.content.res.Configuration

/**
 * Created by Ángel Gómez
 *
 * Contract that defines the Application life cycle used by interceptors.
 */
interface ApplicationLifecycle {

    fun onConfigurationChanged(newConfig: Configuration?)

    fun onCreate()

    fun onLowMemory()

    fun onTrimMemory(level: Int)

}
